pkg/registry/flowcontrol/ensurer/flowschema.go

```diff
@@ -18,191 +18,144 @@ package ensurer
-		wrapper:  wrapper,
-	}
+func (boots *bootstrapFlowSchemas) len() int {
+	return len(boots.boots)
 }
 
-// NewFlowSchemaRemover returns a FlowSchemaRemover instance that
-// can be used to remove a set of FlowSchema configuration objects.
-func NewFlowSchemaRemover(client flowcontrolclient.FlowSchemaInterface, lister flowcontrollisters.FlowSchemaLister) FlowSchemaRemover {
-	return &fsEnsurer{
-		wrapper: &flowSchemaWrapper{
-			client: client,
-			lister: lister,
-		},
+func (boots *bootstrapFlowSchemas) get(i int) bootstrapObject {
+	return &bootstrapFlowSchema{
+		flowSchemaClient: &boots.flowSchemaClient,
+		bootstrap:        boots.boots[i],
 	}
 }
 
-// GetFlowSchemaRemoveCandidates returns a list of FlowSchema object
-// names that are candidates for deletion from the cluster.
-// bootstrap: a set of hard coded FlowSchema configuration objects
-// kube-apiserver maintains in-memory.
-func GetFlowSchemaRemoveCandidates(lister flowcontrollisters.FlowSchemaLister, bootstrap []*flowcontrolv1beta3.FlowSchema) ([]string, error) {
-	fsList, err := lister.List(labels.Everything())
+func (boots *bootstrapFlowSchemas) getExistingObjects() ([]deletable, error) {
+	objs, err := boots.lister.List(labels.Everything())
 	if err != nil {
-		return nil, fmt.Errorf("failed to list FlowSchema - %w", err)
+		return nil, fmt.Errorf("failed to list FlowSchema objects - %w", err)
 	}
-
-	bootstrapNames := sets.String{}
-	for i := range bootstrap {
-		bootstrapNames.Insert(bootstrap[i].GetName())
-	}
-
-	currentObjects := make([]metav1.Object, len(fsList))
-	for i := range fsList {
-		currentObjects[i] = fsList[i]
+	dels := make([]deletable, len(objs))
+	for i, obj := range objs {
+		dels[i] = &deletableFlowSchema{
+			FlowSchema: obj,
+			client:     boots.client,
+		}
 	}
-
-	return getDanglingBootstrapObjectNames(bootstrapNames, currentObjects), nil
+	return dels, nil
 }
 
-type fsEnsurer struct {
-	strategy ensureStrategy
-	wrapper  configurationWrapper
+type bootstrapFlowSchema struct {
+	*flowSchemaClient
+	bootstrap *flowcontrolv1beta3.FlowSchema
 }
 
-func (e *fsEnsurer) Ensure(flowSchemas []*flowcontrolv1beta3.FlowSchema) error {
-	for _, flowSchema := range flowSchemas {
-		if err := ensureConfiguration(e.wrapper, e.strategy, flowSchema); err != nil {
-			return err
-		}
-	}
+func (boot *bootstrapFlowSchema) getName() string {
+	return boot.bootstrap.Name
+}
 
-	return nil
+func (boot *bootstrapFlowSchema) create(ctx context.Context) error {
+	_, err := boot.client.Create(ctx, boot.bootstrap, metav1.CreateOptions{FieldManager: fieldManager})
+	return err
 }
 
-func (e *fsEnsurer) RemoveAutoUpdateEnabledObjects(flowSchemas []string) error {
-	for _, flowSchema := range flowSchemas {
-		if err := removeAutoUpdateEnabledConfiguration(e.wrapper, flowSchema); err != nil {
-			return err
-		}
+func (boot *bootstrapFlowSchema) getCurrent() (wantAndHave, error) {
+	current, err := boot.lister.Get(boot.bootstrap.Name)
+	if err != nil {
```

Comment:
Don't we want to handle `NotFound` error differently here?

Reply:
The caller handles the NotFound error.  The `wantAndHave` interface does not have a decent interpretation in the NotFound case.